src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/HttpApi/ApiClient.cs

```diff
@@ -304,10 +304,53 @@ public async Task<string> GetMetricsAsync(CancellationToken token)
             throw await CreateUnexpectedStatusCodeExceptionAsync(response).ConfigureAwait(false);
         }
 
+        public async Task<HttpResponseMessage> ApiCall(string routeAndQuery, CancellationToken token)
+        {
+            using HttpRequestMessage request = new(HttpMethod.Get, routeAndQuery);
+            HttpResponseMessage response = await SendAndLogAsync(request, HttpCompletionOption.ResponseHeadersRead, token).ConfigureAwait(false);
+            return response;
+        }
+
+        public async Task<OperationResponse> EgressTraceAsync(int processId, int durationSeconds, string egressProvider, CancellationToken token)
+        {
+            string uri = FormattableString.Invariant($"/trace?pid={processId}&egressProvider={egressProvider}&durationSeconds={durationSeconds}");
+            using HttpRequestMessage request = new(HttpMethod.Get, uri);
+            using HttpResponseMessage response = await SendAndLogAsync(request, HttpCompletionOption.ResponseContentRead, token).ConfigureAwait(false);
+
+            return new OperationResponse(response.StatusCode, response.Headers.Location, await response.Content.ReadAsStringAsync().ConfigureAwait(false));
```

Comment:
This method should be enforcing successful before returning, something like:

```cs
switch (response.StatusCode)
{
    case HttpStatusCode.OK:
        ValidateContentType(response, ContentTypes.ApplicationJson);
        return new OperationResponse(response.StatusCode, response.Headers.Location, await response.Content.ReadAsStringAsync().ConfigureAwait(false));
    case HttpStatusCode.BadRequest:
    case HttpStatusCode.TooManyRequests:
        ValidateContentType(response, ContentTypes.ApplicationProblemJson);
        throw await CreateValidationProblemDetailsExceptionAsync(response).ConfigureAwait(false);        
    case HttpStatusCode.Unauthorized:
        ThrowIfNotSuccess(response);
        break;
}

throw await CreateUnexpectedStatusCodeExceptionAsync(response).ConfigureAwait(false);
```

and the test callers should be catching the exceptions when those are expected (e.g. when testing the throttling).

All of the new ApiClient methods should be do this.